src/config.py

```diff
@@ -0,0 +1,348 @@
+
+        :return: bool
+            True if the environment is a production environment, False otherwise.
+            It will also return False if not running on Platform.sh or in the build phase.
+
+        """
+
+        if not self.is_valid_platform() and not self.in_build():
+            return False
+        prod_branch = 'production' if self.on_enterprise() else 'master'
+        return self._get_value('BRANCH') == prod_branch
+
+    def __get_value(self, name):
+        """Reads an environment variable, taking the prefix into account.
+
+        :param name: string
+            The variable to read.
+        :return: string|None
+        """
+
+        check_name = self.envPrefix + name.upper()
+        return self.environmentVariables[check_name] if check_name in self.environmentVariables.keys() else None
+
+    @staticmethod
+    def decode(variable):
+        """Decodes a Platform.sh environment variable.
+
+        :param variable: string
+            Base64-encoded JSON (the content of an environment variable).
+        :exception:
+            JSON decoding error.
+        :return: mixed
+            An associative array (if representing a JSON object), or a scalar type.
+
+        """
+
+        try:
+            # if sys.version_info[1] > 5:
+            return json.loads(base64.b64decode(variable))
+                # return json.loads(variable.decode("utf-8"))
+            # else:
+            #     return json.loads(base64.decodestring(variable))
+        except json.decoder.JSONDecodeError:
+            print('Error decoding JSON, code %d', json.decoder.JSONDecodeError)
+
+    def __getattr__(self, config_property):
+        """Gets a configuration property.
+
+        :param config_property: string
+            A (magic) property name. The properties are documented in the DocBlock for this class.
+        :exception:
+            If a variable is not found, or if decoding fails.
+        :return: mixed
+            The return types are documented in the DocBlock for this class.
+
+        """
+
+        if not self.is_valid_platform():
+            raise RuntimeError('You are not running on Platform.sh, so the {} variable is '
+                               'not available.'.format(config_property))
+        is_build_var = config_property in self.directVariables.keys()
+        is_runtime_var = config_property in self.directVariablesRuntime.keys()
+        if self.in_build() and is_runtime_var:
+            raise ValueError('The {} variable is not available during build time.'.format(config_property))
+        if is_build_var:
+            return self._get_value(self.directVariables[config_property])
+        if is_runtime_var:
+            return self._get_value(self.directVariablesRuntime[config_property])
+        raise ValueError('No such variable defined: '.format(config_property))
+
+    def isset(self, config_property):
+        """Checks whether a configuration property is set.
+
+        :param config_property:
```

Comment:
Just FYI, if you do implement `__getitem__` above as I suggested, you should also implement `__contains__`, which can just call this method directly.

Reply:
@chadwcarlson On second thought, since `__getattr__` and `__getitem__` are going to return different things, don't define `__contains__`. By default, if you don't have a `__contains__`, if a user says something like `if project in config`, it will call `__getitem__` and see if it raises an error. This is what we want since our properties are going to be accessed as attributes.

Someone using this library should know to use `hasattr(config, project)`, for example, which will try to `getattr` that attribute and return true or false depending on if `getattr` raises an exception